pytest_reportportal/service.py

```diff
@@ -529,14 +529,18 @@ def _add_item_hier_parts_other(item_parts, item, item_type, hier_flag,
         """
         for part in item_parts:
 
-            if type(part) is item_type:
+            if isinstance(part, item_type):
 
                 if item_type is Module:
                     module_path = str(
                         item.fspath.new(dirname=rp_name,
                                         basename=part.fspath.basename,
                                         drive=""))
-                    rp_name = module_path if rp_name else module_path[1:]
+                    if module_path.startswith(':\\', 1):
+                        stripped_module_path = module_path[3:]
```

Comment:
Can you please try not to use slicing? Search through the os.path for better solutions.